hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieDropPartitionsTool.java

```diff
@@ -0,0 +1,537 @@
+    sb.append("   --hive-partition-value-extractor-class ").append(cfg.partitionValueExtractorClass).append(", \n");
+    sb.append("   --spark-master ").append(cfg.sparkMaster).append(", \n");
+    sb.append("   --spark-memory ").append(cfg.sparkMemory).append(", \n");
+    sb.append("   --props ").append(cfg.propsFilePath).append(", \n");
+    sb.append("   --hoodie-conf ").append(cfg.configs);
+    sb.append("\n}");
+    return sb.toString();
+  }
+
+  public static void main(String[] args) {
+    final Config cfg = new Config();
+    JCommander cmd = new JCommander(cfg, null, args);
+    if (cfg.help || args.length == 0) {
+      cmd.usage();
+      System.exit(1);
+    }
+    SparkConf sparkConf = UtilHelpers.buildSparkConf("Hoodie-Drop-Table-Partitions", cfg.sparkMaster);
+    sparkConf.set("spark.executor.memory", cfg.sparkMemory);
+    JavaSparkContext jsc = new JavaSparkContext(sparkConf);
+    HoodieDropPartitionsTool tool = new HoodieDropPartitionsTool(jsc, cfg);
+    try {
+      tool.run();
+    } catch (Throwable throwable) {
+      LOG.error("Fail to run deleting table partitions for " + tool.getConfigDetails(), throwable);
+    } finally {
+      jsc.stop();
+    }
+  }
+
+  public void run() {
+    try {
+      if (StringUtils.isNullOrEmpty(cfg.instantTime)) {
+        cfg.instantTime = HoodieActiveTimeline.createNewInstantTime();
+      }
+      LOG.info(getConfigDetails());
+
+      Mode mode = Mode.valueOf(cfg.runningMode.toUpperCase());
+      switch (mode) {
+        case DELETE_PARTITIONS_LAZY:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in delete_partition_lazy mode ******");
+          doDeleteTablePartitionsLazy();
+          break;
+        case DELETE_PARTITIONS_EAGER:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in delete_partition_eager mode ******");
+          doDeleteTablePartitionsEager();
+          break;
+        case DELETE_PARTITIONS_QUIET:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in delete_partition_quiet mode ******");
+          doDeleteTablePartitionsQuiet();
+          break;
+        case DRY_RUN:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in dry-run mode ******");
+          dryRun();
+          break;
+        default:
+          LOG.info("Unsupported running mode [" + cfg.runningMode + "], quit the job directly");
+      }
+    } catch (Exception e) {
+      throw new HoodieException("Unable to delete table partitions in " + cfg.basePath, e);
+    }
+  }
+
+  public void doDeleteTablePartitionsLazy() {
+    doDeleteTablePartitions();
+    syncToHiveIfNecessary(false);
+  }
+
+  public void doDeleteTablePartitionsEager() {
+    doDeleteTablePartitions();
+    deleteDataFiles();
+    deleteEmptyDirIfNecessary();
+    syncToHiveIfNecessary(false);
+  }
+
+  public void doDeleteTablePartitionsQuiet() {
+
+    deleteDataFiles();
```

Comment:
same comment as above. Chances that metadata table will be out of sync. so we need to cautious in doing these.